File: server/models/menuItem.model.js
import mongoose from 'mongoose';

const menuItemSchema = new mongoose.Schema({
  id: { type: String, required: true, unique: true },
  name: { type: String, required: true },
  description: String,
  price: { type: Number, required: true },
  imageUrl: String,
  isTrending: { type: Boolean, default: false },
  isAvailable: { type: Boolean, default: true },
  category: String,
},{ collection: 'items' });

export const MenuItem = mongoose.model('MenuItem', menuItemSchema);

File: client/lib/features/menu/data/models/menu_item_model.dart
import '../../domain/entities/menu_item_entity.dart';

class MenuItemModel extends MenuItemEntity {
  MenuItemModel({
    required String id,
    required String name,
    required String description,
    required double price,
    required String imageUrl,
    bool isTrending = false,
    required bool isAvailable,
    required String? category,
  }) : super(
         id: id,
         name: name,
         description: description,
         price: price,
         imageUrl: imageUrl,
         isTrending: isTrending,
         isAvailable: isAvailable,
       );

  factory MenuItemModel.fromJson(Map<String, dynamic> json) {
    return MenuItemModel(
      id: json['id'],
      name: json['name'],
      description: json['description'],
      price: (json['price'] as num).toDouble(),
      imageUrl: json['imageUrl'],
      isTrending: json['isTrending'] ?? false,
      isAvailable: json['isAvailable'] ?? false,
      category: json['category'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'imageUrl': imageUrl,
      'isTrending': isTrending,
      'isAvailable': isAvailable,
      'category': category,
    };
  }
}
